### PR TITLE
UML-1886: cookies page updated and banner shown on setting cookie preferences

### PR DESCRIPTION
**Pull request by SeemaMenon**
# Purpose

https://opgtransform.atlassian.net/browse/UML-1886

Fixes UML-1866

## Approach

_Explain how your code addresses the purpose of the change_

## Learning

_Any tips and tricks, blog posts or tools which helped you. Plus anything notable you've discovered about the Use a Lasting Power of Attorney service_

## Checklist

* [X] I have performed a self-review of my own code
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] New event_codes have been documented on the [wiki page](https://opgtransform.atlassian.net/wiki/spaces/LSML2/pages/3277881441/Understanding+the+event+logs)
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [X] I have added tests to prove my work
* [x] I have added welsh translation tags and updated translation files
* [ ] I have run an accessibility tool on any pages I have made changes to and fixed any issues found
* [ ] I have notified the Interaction Designer of any content changes so that appropriate screenshots/flow diagram changes can be made
* [ ] The product team have tested these changes

### CODE REVIEW

**Comment by SeemaMenon** [comment]
**Changes:**
1. Cookies page content updated.
2. On changing cookie preferences, now shows a success banner in the cookies page.
3. The link in the banner, takes user back to the page they were previously on.
[Above applicable for actor service and viewer service.]

Screenshot 2022-04-14 at 18 13 09
Screenshot 2022-04-14 at 18 13 25
Screenshot 2022-04-14 at 18 17 34

File: service-front/app/languages/cy/LC_MESSAGES/messages.po [hunk 1]
```diff
@@ -6678,3 +6679,177 @@ msgstr "Gweld y cwcis"
...
+#: src/Common/templates/common/cookies.html.twig:22
+msgid ""
+"Cookies are small files saved on your phone, tablet or computer when you visit "
+"a website."
+msgstr ""
+"Ffeiliau a arbedir ar eich ffôn, eich tabled neu eich cyfrifiadur pan fyddwch "
+"yn ymweld â gwefan yw cwcis."
+
+#: src/Common/templates/common/cookies.html.twig:26
+#| msgid ""
```
Comment (lukejolliffe) [inline service-front/app/languages/cy/LC_MESSAGES/messages.po:6846]: This comment can be removed

Comment (lukejolliffe) [reply service-front/app/languages/cy/LC_MESSAGES/messages.po:6846]: I would just quickly double check the welsh is right here as well, sometimes this is poedit trying to be clever

**Review by lukejolliffe — CHANGES_REQUESTED** [review]

**Review by cooperaj — CHANGES_REQUESTED** [review]

File: service-front/app/features/context/UI/CommonContext.php [hunk 2]
```diff
@@ -115,6 +115,9 @@ public function iAttachATracingHeaderToMyRequests()
     public function iCanSeeACookieConsentBanner($serviceName)
     {
         $this->ui->assertPageAddress('/home');
+
+        //var_dump($serviceName); die;
```
[the remark this answers is not in the document]
Comment (SeemaMenon) [reply service-front/app/features/context/UI/CommonContext.php:119]: My bad..... removed now.

File: service-front/app/features/context/UI/CommonContext.php [hunk 3]
```diff
@@ -218,17 +221,28 @@ public function iExpectToBeOnTheGovUkHomepage()
...
     }
 
+    /**
+     * @Then /^I am shown cookie preferences has been set$/
+     */
+    public function iAmShownCookiePreferencesHasBeenSet()
+    {
+        $this->ui->assertPageAddress('/cookies');
+        $this->ui->assertPageContainsText(
+            "You’ve set your cookie preferences. Go back to the page you were looking at."
```
[the remark this answers is not in the document]
Comment (SeemaMenon) [reply service-front/app/features/context/UI/CommonContext.php:241]: Currently it takes the user back to the previous page automatically.


How this story says user to stay on the cookie page with a banner on changing cookie preferences.

`**Acceptance criteria**

- success banner in attachment shown when user marks their options and when they click the ‘save changes’ button on the page

- Banner appears on the same cookie page

- When user clicks on ‘[Go back to the page you were looking a](https://design-system.service.gov.uk/patterns/cookies-page/cookies-updated/index.html#)t’, they need to be taken to the previous page they were on`

File: service-front/app/src/Common/src/Handler/CookiesPageHandler.php [hunk 4]
```diff
@@ -73,19 +85,28 @@ public function handle(ServerRequestInterface $request): ResponseInterface
...
 
-        $response = new RedirectResponse(
-            $this->urlValidityCheckService->setValidReferer($form->get('referer')->getValue())
+        $response = new RedirectResponse($this->urlHelper->generate('cookies'));
+
+        /** @var FlashMessagesInterface $flash */
+        $flash = $request->getAttribute(FlashMessageMiddleware::FLASH_ATTRIBUTE);
+
+        $message = $this->translator->translate(
+            "%referer%",
```
[the remark this answers is not in the document]
Comment (SeemaMenon) [reply service-front/app/src/Common/src/Handler/CookiesPageHandler.php:104]: True...nothing, so I removed this line and passing in the referer as a parameter to flash
` $flash->flash(self::COOKIES_SET_FLASH_MSG, $form->get('referer')->getValue());`

File: service-front/app/src/Common/templates/partials/flash-message.html.twig [hunk 5]
```diff
@@ -0,0 +1,15 @@
+{% if flash.getFlash(constant(flash_key)) %}
+    <div class="govuk-notification-banner govuk-notification-banner--success" role="alert" aria-labelledby="govuk-notification-banner-title" data-module="govuk-notification-banner">
+        <div class="govuk-notification-banner__header">
+            <h2 class="govuk-notification-banner__title" id="govuk-notification-banner-title">
+                Success
+            </h2>
+        </div>
+        <div class="govuk-notification-banner__content">
+            <p class="govuk-notification-banner__heading">
+                {% trans %}You’ve set your cookie preferences.{% endtrans %}
```
[the remark this answers is not in the document]
Comment (SeemaMenon) [reply service-front/app/src/Common/templates/partials/flash-message.html.twig:10]: That's right.... I wasn't sure how to get the url [to take user back to the previous page] in the flash message. 

This user story is likely to get revised [a new ticket in making by Sarah], and the flash message is likely to change to the previous page used was on....like current pattern.

Can this stay until we get to that ticket?

**Comment by SeemaMenon** [comment]
> Mmm. Is it not possible to automatically redirect the user to their referring page and show the appropriate flash message there?
> 
> Though I think that may need a rework of the flash messages in general.

This user story is likely to get revised [a new ticket in making by Sarah], and the flash message is likely to change to the previous page used was on....like current pattern.

File: .idea/runConfigurations/Front_UI.xml [hunk 6]
```diff
@@ -1,11 +0,0 @@
-<component name="ProjectRunConfigurationManager">
```
Comment (cooperaj) [inline .idea/runConfigurations/Front_UI.xml:1]: Wait, why is this deleted?

**Review by cooperaj — APPROVED** [review]

**Review by lukejolliffe — APPROVED** [review]

**Pull request merged**